src/Factories/SearchParametersFactory.php

```diff
@@ -77,6 +77,18 @@ protected function makeFilter(Builder $builder): ?array
             $filter = $filter->merge($whereIns);
         }
 
+        if (property_exists($builder, 'whereNotIns')) {
+            $whereNotIns = collect($builder->whereNotIns)->map(static fn (array $values, string $field) => [
+                'terms' => [$field => $values],
+            ])->values();
+
+            $filter = $filter->merge([[
```

Comment:
Perhaps we can just use `$filter->push()` to avoid double `[` brackets?

Can I also ask you to add tests (see `SearchParametersFactoryTest::test_search_parameters_can_be_made_from_builder_with_wherein_filter()` and `EngineSearchTest::test_search_result_can_be_filtered_with_wherein_clause()` for the reference)?